File: frontend/src/api/gameData.ts
import type { Item, Recipe, Building, Generator } from '../types/gameData';
import { apiFetch } from './client';

export const fetchItems = () => apiFetch<Item[]>('/items');
export const fetchRecipes = () => apiFetch<Recipe[]>('/recipes');
export const fetchBuildings = () => apiFetch<Building[]>('/buildings');
export const fetchGenerators = () => apiFetch<Generator[]>('/generators');

File: backend/src/routes/game_data.rs
use axum::{extract::State, Json};

use crate::models::game_data::{Building, Item, Recipe};
use crate::state::AppState;

pub async fn get_items(State(state): State<AppState>) -> Json<Vec<Item>> {
    Json(state.game_data.items.clone())
}

pub async fn get_recipes(State(state): State<AppState>) -> Json<Vec<Recipe>> {
    Json(state.game_data.recipes.clone())
}

pub async fn get_buildings(State(state): State<AppState>) -> Json<Vec<Building>> {
    Json(state.game_data.buildings.clone())
}
